### PR TITLE
Port over doc comments in route macros.

### LINKED ISSUE #1389: proc-macro docstrings issue
As discussed on gitter (https://gitter.im/actix/actix?at=5e58817aca2f5a558d544542), it appears that docstrings used with the proc-macros gets wrapped into the inner function.

An example of the problem:
```rust
// Index function
///
/// Arguements:
///
/// * None
///
/// Returns:
/// * Responder with the string `Hello World!` in it.
#[get("/")]
pub async fn index() -> impl Responder {
    "Hello World!".to_string()
}
```

Then when running `cargo doc` on the project, the above docstring does not get rendered.

### PR DESCRIPTION
**Pull request by erikjohnston**
## PR Type

Bug Fix / Feature

## PR Checklist

- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.
- [x] A changelog entry has been made for the appropriate packages.
- [x] Format code with the latest stable rustfmt.
- [x] (Team) Label with affected crates and semver status.

## Overview

This allows documentation on the functions to appear in the generated
docs, e.g.:

```rust
/// The index page
async fn index() -> &'static str {
    "Hello!"
}
```

Closes #1389

--- 

I'm afraid I have no idea how to add tests for this beyond making sure this doesn't break when using docstrings.

### CODE REVIEW

**Comment by robjtede** [comment]
Does this handle multi line doc-comments? I don't recall if they're converted to one doc attribute with line breaks or multiple doc attrs.

**Comment by erikjohnston** [comment]
> Does this handle multi line doc-comments? I don't recall if they're converted to one doc attribute with lie breaks or multiple doc attrs.

Huh, you're right, had no idea that is multi line comments worked. I've pushed a fix and manually tested.

Tangentially, I am wondering whether it'd be better to just copy all attributes over? It's probably not *that* useful, but would mean e.g. that people could have manually added `#[allow(missing_docs)]` to the structs.

**Comment by robjtede** [comment]
> Tangentially, I am wondering whether it'd be better to just copy all attributes over?

I'm apprehensive about that because it may seem strange that attributes meant for functions will not work on the generated struct. If you don't feel strongly that it should be done I'd like to wait until it is raised specifically.

File: actix-web-codegen/src/route.rs [hunk 1]
```diff
@@ -221,6 +224,20 @@ impl Route {
...
 
+        // Try and pull out the doc comments so that we can reapply them to the
+        // generated struct.
+        //
+        // Note that multi line doc comments are converted to multiple doc
+        // attributes.
+        let mut doc_attribute = Vec::new();
+        for attr in &ast.attrs {
+            if let Some(ident) = attr.path.get_ident() {
+                if ident == "doc" {
```
Comment (jplatte) [inline actix-web-codegen/src/route.rs]:
```suggestion
            if attr.path.get_ident().is_ident("doc") {
```

File: actix-web-codegen/src/route.rs [hunk 2]
```diff
@@ -221,6 +224,20 @@ impl Route {
         let ast: syn::ItemFn = syn::parse(input)?;
         let name = ast.sig.ident.clone();
 
+        // Try and pull out the doc comments so that we can reapply them to the
+        // generated struct.
+        //
+        // Note that multi line doc comments are converted to multiple doc
+        // attributes.
+        let mut doc_attribute = Vec::new();
```
Comment (jplatte) [inline actix-web-codegen/src/route.rs]:
```suggestion
        let mut doc_attributes = Vec::new();
```

then the return value can also use struct init shorthand for this.

Comment (jplatte) [reply actix-web-codegen/src/route.rs]: Actually I think this + the whole loop can be simplified to

```rust
let doc_attributes = ast.attrs.iter().filter(|attr| attr.path.is_ident("doc")).cloned().collect();
```

File: actix-web-codegen/src/route.rs [hunk 3]
```diff
@@ -265,6 +283,7 @@ impl ToTokens for Route {
                     methods,
                 },
             resource_type,
+            doc_attributes: doc_attribute,
```
Comment (jplatte) [inline actix-web-codegen/src/route.rs]:
```suggestion
            doc_attributes,
```

File: actix-web-codegen/src/route.rs [hunk 4]
```diff
@@ -286,7 +305,13 @@ impl ToTokens for Route {
             }
         };
 
+        let doc_comment_token_stream: TokenStream2 = doc_attribute
+            .iter()
+            .map(ToTokens::to_token_stream)
+            .collect();
+
```
Comment (jplatte) [inline actix-web-codegen/src/route.rs]: This is not needed, you can just use a repetition in the output. Suggestion in next review comment.

File: actix-web-codegen/src/route.rs [hunk 5]
```diff
@@ -286,7 +305,13 @@ impl ToTokens for Route {
...
         let stream = quote! {
+            #doc_comment_token_stream
```
Comment (jplatte) [inline actix-web-codegen/src/route.rs]:
```suggestion
            #(#doc_comments)*
```

File: actix-web-codegen/src/route.rs (hunk 1 above)
Comment (robjtede) [reply actix-web-codegen/src/route.rs]: https://docs.rs/syn/1.0.60/syn/struct.Path.html#method.is_ident

**Comment by erikjohnston** [comment]
Thank you both, my proc macro foo is still quite weak :heart:

**Review by jplatte — APPROVED** [review]

**Review by robjtede — APPROVED** [review]

**Pull request merged**